spring-xsuaa/src/test/java/com/sap/cloud/security/xsuaa/token/flows/UserTokenFlowTests.java

```diff
@@ -0,0 +1,301 @@
+					TestConstants.tokenEndpointUri, TestConstants.authorizeEndpointUri,
+					TestConstants.keySetEndpointUri);
+			tokenFlow.token(validMockJwt)
+					.execute();
+		}).isInstanceOf(TokenFlowException.class).hasMessageContaining("User token flow request is not valid");
+
+		assertThatThrownBy(() -> {
+			UserTokenFlow tokenFlow = new UserTokenFlow(restTemplate, refreshTokenFlowMock,
+					TestConstants.tokenEndpointUri, TestConstants.authorizeEndpointUri,
+					TestConstants.keySetEndpointUri);
+			tokenFlow.client(clientId)
+					.secret(clientSecret)
+					.execute();
+		}).isInstanceOf(TokenFlowException.class).hasMessageContaining("User token not set");
+
+		assertThatThrownBy(() -> {
+			UserTokenFlow tokenFlow = new UserTokenFlow(restTemplate, refreshTokenFlowMock,
+					TestConstants.tokenEndpointUri, TestConstants.authorizeEndpointUri,
+					TestConstants.keySetEndpointUri);
+			tokenFlow.client(null)
+					.secret(clientSecret)
+					.execute();
+		}).isInstanceOf(IllegalArgumentException.class).hasMessageContaining("client ID");
+
+		assertThatThrownBy(() -> {
+			UserTokenFlow tokenFlow = new UserTokenFlow(restTemplate, refreshTokenFlowMock,
+					TestConstants.tokenEndpointUri, TestConstants.authorizeEndpointUri,
+					TestConstants.keySetEndpointUri);
+			tokenFlow.client(clientId)
+					.secret(null)
+					.execute();
+		}).isInstanceOf(IllegalArgumentException.class).hasMessageContaining("client secret");
+	}
+
+	@Test
+	public void test_execute() throws TokenFlowException {
+
+		URI expectedRequestURI = UriComponentsBuilder.fromUri(TestConstants.tokenEndpointUri)
+				.queryParam("grant_type", "user_token")
+				.queryParam("response_type", "token")
+				.queryParam("client_id", clientId)
+				.build().toUri();
+
+		HttpEntity<Void> expectedRequest = buildExpectedRequest();
+
+		RestTemplateMock restTemplateMock = new RestTemplateMock(expectedRequestURI, expectedRequest, Map.class,
+				validMockJwt.getTokenValue(), HttpStatus.OK);
+
+		UserTokenFlow tokenFlow = new UserTokenFlow(restTemplateMock, refreshTokenFlowMock,
+				TestConstants.tokenEndpointUri, TestConstants.authorizeEndpointUri, TestConstants.keySetEndpointUri);
+		tokenFlow.token(validMockJwt)
+				.client(clientId)
+				.secret(clientSecret)
+				.execute();
+
+		restTemplateMock.validateCallstate();
+		refreshTokenFlowMock.validateCallstate();
+	}
+
+	@Test
+	public void test_execute_throwsIfHttpStatusUnauthorized() throws TokenFlowException {
+
+		URI expectedRequestURI = UriComponentsBuilder.fromUri(TestConstants.tokenEndpointUri)
+				.queryParam("grant_type", "user_token")
+				.queryParam("response_type", "token")
+				.queryParam("client_id", clientId)
+				.build().toUri();
+
+		HttpEntity<Void> expectedRequest = buildExpectedRequest();
+
+		RestTemplateMock restTemplateMock = new RestTemplateMock(expectedRequestURI, expectedRequest, Map.class,
+				validMockJwt.getTokenValue(), HttpStatus.UNAUTHORIZED);
```

Comment:
need to fix this test as an unauthorized is represented by a HttpClientErrorException

Reply:
Done